src/rpc/blockchain.cpp

```diff
@@ -1032,128 +1032,250 @@ UniValue reconsiderblock(const UniValue& params, bool fHelp)
+            "     \"constant\":                xxxxx,   (string)  The arbitrary byte string of constant set at sc creation\n"
+            "     \"immature amounts\": [\n"
+            "       {\n"
+            "         \"maturityHeight\":      xxxxx,   (numeric) height at which fund will become part of spendable balance\n"
+            "         \"amount\":              xxxxx,   (numeric) immature fund\n"    
+            "       },\n"
+            "       ... ]\n"
+            "    },\n"
+            "    ...\n"
+            "  ]\n"
+            "}\n"
 
             "\nExamples\n"
             + HelpExampleCli("getscinfo", "\"1a3e7ccbfd40c4e2304c3215f76d204e4de63c578ad835510f580d529516a874\"")
-            + HelpExampleCli("getscinfo", "")
+            + HelpExampleCli("getscinfo", "\"*\" true false 2 10")
+            + HelpExampleCli("getscinfo", "\"*\" ")
         );
 
-    if (params.size() > 0)
+    bool bRetrieveAllSc = false;
+    string inputString = params[0].get_str();
+    if (!inputString.compare("*"))
+    	bRetrieveAllSc = true;
+    else
     {
-        // side chain id
-        string inputString = params[0].get_str();
         if (inputString.find_first_not_of("0123456789abcdefABCDEF", 0) != std::string::npos)
             throw JSONRPCError(RPC_TYPE_ERROR, "Invalid scid format: not an hex");
- 
+    }
+
+    bool bOnlyAlive = false;
+    if (params.size() > 1)
+    	bOnlyAlive = params[1].get_bool();
+
+    bool bVerbose = true;
+    if (params.size() > 2)
+    	bVerbose = params[2].get_bool();
+
+    UniValue ret(UniValue::VOBJ);
+    UniValue scItems(UniValue::VARR);
+
+    if (!bRetrieveAllSc)
+    {
+        // single search
         uint256 scId;
         scId.SetHex(inputString);
  
-        UniValue sc(UniValue::VOBJ);
-        if (!AddScInfoToJSON(scId, sc) )
+        UniValue scRecord(UniValue::VOBJ);
+        // throws a json rpc exception if the scid is not found in the db
+        if (!FillScRecord(scId, scRecord, bOnlyAlive, bVerbose) )
         {
-            throw JSONRPCError(RPC_INVALID_PARAMETER, string("scid not yet created: ") + scId.ToString());
+            // after filtering no sc has been found, this can happen for instance when the sc is ceased
+            // and bOnlyAlive is true
+            ret.push_back(Pair("totalItems", 0));
+            ret.push_back(Pair("from", 0));
+            ret.push_back(Pair("to", 0));
+        }
+        else
+        {
+            ret.push_back(Pair("totalItems", 1));
+            ret.push_back(Pair("from", 0));
+            ret.push_back(Pair("to", 1));
+            scItems.push_back(scRecord);
         }
- 
-        return sc;
     }
+    else
+    {
+        int from = 0;
+        if (params.size() > 3)
+    	    from = params[3].get_int();
 
-    // dump all of them if any
-    UniValue result(UniValue::VARR);
-    AddScInfoToJSON(result);
+        int to = 50;
```

Comment:
as discussed AFK, `to` default should be `-1`

Reply:
Done